remote-messaging/remote-messaging-store/src/test/java/com/duckduckgo/remote/messaging/store/RemoteMessagingCohortStoreDBTest.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.remote.messaging.store
+
+import androidx.room.Room
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import org.junit.Assert.*
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class RemoteMessagingCohortStoreDBTest {
+
+    private val db = Room.inMemoryDatabaseBuilder(InstrumentationRegistry.getInstrumentation().targetContext, RemoteMessagingDatabase::class.java)
+        .allowMainThreadQueries()
+        .build()
+    private val cohortDao = db.remoteMessagingCohortDao()
+    private val testee = RemoteMessagingCohortStoreDB(db)
+
+    @Test
+    fun whenPercentileIsSetForMessageThenReturnSameValue() {
+        cohortDao.insert(RemoteMessagingCohort(message = "message1", percentile = 0.5f))
+        assertEquals(0.5f, testee.getPercentile("message1"))
+    }
+
```

Comment:
oh, good catch with this one. I missed the `autoGenerate = true` in the db entity. However, checking previous comments, I agree we move to `messageId` as primary key here.

Reply:
We also need to see how to do so that the table doesn't grow unbounded (?)

Reply:
For completeness it's something I want to include. Easy approach I've been thinking about was to delete all messageIds not present in the config. But I haven't spent enough time thinking about pros/cons of doing that, or if we should do something different. Not something to include now, since that table is going to grow really slow.